### ISSUE TITLE
unable to execute /usr/bin/find: Argument list too long

### ISSUE THREAD

**Issue by nmiculinic** [CyberShadow/aconfmgr#17]
After running aconfmgr apply following error appears:
```
: Collecting data...
:: Compiling user configuration...
::: Using configuration in /home/lpp/.config/aconfmgr
::: Sourcing /home/lpp/.config/aconfmgr/10-ignores.sh...
::: Done (0 native packages, 0 foreign packages, 0 files).
:: Inspecting system state...
::: Querying package list...
:::: Done.
::: Enumerating managed files...
:::: Done.
::: Searching for lost files...
sudo: unable to execute /usr/bin/find: Argument list too long
:::: Fatal error! Stack trace:
::::: /usr/lib/aconfmgr/common.bash:1 [AconfCompileSystem]
::::: /usr/lib/aconfmgr/common.bash:452 [AconfCompile]
::::: /usr/lib/aconfmgr/save.bash:9 [AconfSave]
::::: /usr/bin/aconfmgr:164 [Main]
::::: /usr/bin/aconfmgr:178 [main]
```

**Comment by CyberShadow** [comment]
This should only happen if you have a *lot* of ignored files (total size of find's command line is greater than your system's limit). Do you have a lot of `IgnorePath` directives in your configuration?

You should be able to work around this by reducing the number of `IgnorePath` directives (e.g. by grouping several entries into one using wildcards), or increasing your system's `ARG_MAX` setting.

Unfortunately there doesn't seem to be a simple way to work around this from `aconfmgr`, since GNU find does not seem to support reading the expression from a file.

**Comment by nmiculinic** [comment]
Hmm.. I guess 
```IgnorePath /etc/ca-certificates``` is enough and ```IgnorePath /etc/ca-certificates/*``` presents problems. I'm closing.

**Comment by CyberShadow** [comment]
> IgnorePath /etc/ca-certificates/*

That doesn't do what you mean - it will expand the list of file names when your config file is parsed. Quote the `IgnorePath` argument to prevent globbing.

**Comment by nmiculinic** [comment]
Didn't know that, thanks. Is there any difference between those two IgnorePath commands?

**Comment by CyberShadow** [comment]
Ah, I see the documentation was misleading. Fixed in aeebd9abdf26bf20a44bc974e05350bdaa61e623

> Didn't know that, thanks. Is there any difference between those two IgnorePath commands?

Without quoting, the globbing is done when the configuration is parsed, meaning that `IgnorePath` sees the full file list as it is present on your system. With quoting, the pattern is passed to `find` for it to match against files it finds.

All things considered, in most cases, there wouldn't be much of a difference, however if the argument is not quoted and some matched file names have special characters, they would be interpreted again by `find` as shell patterns, which might result in unwanted behavior. And, of course, the issue you ran into - that the expanded list is too long to be passed to `find`.
